### PR TITLE
chore: remove outdated release instructions

### PR DESCRIPTION
**Pull request by Rustin170506**
## Motivation

Follow up of https://github.com/tokio-rs/tracing/commit/60c60bef62972e447414a748a95b31ff9027165b

## Solution

I have removed outdated release instructions from our document.

### CODE REVIEW

**Review by hds — APPROVED** [review]
As @kaffarell said, we don't have links in most of the Cargo.toml files and some of the README.md files don't have versioned links anymore either.

If you can fix those, that would be great. But I'll take this change as it is as a net positive, and I'm happy to merge as is.

File: tracing-attributes/Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,6 @@
 name = "tracing-attributes"
 # When releasing to crates.io:
 # - Remove path dependencies
-# - Update html_root_url.
 # - Update doc url
 #   - Cargo.toml
 #   - README.md
```
[the remark this answers is not in the document]
Comment (Rustin170506) [reply tracing-attributes/Cargo.toml]: Good catch! Updated.

**Comment by Rustin170506** [comment]
@hds I have updated it. Thank you!

**Comment by hds** [comment]
@Rustin170506 Thanks! Merged.

**Pull request merged**